// my/express/1.0.0/my.express-1.0.0.js
(function(define) { define(function(require, ex$, module) {
var $$req$ = require; require = (typeof $$ceylon$require == 'undefined') ? $$req$ : function() { return $$ceylon$require('my/express/1.0.0/my.express-1.0.0', $$req$, Array.prototype.slice.call(arguments)); }

var _CTM$;function $CCMM$(){if (_CTM$===undefined)_CTM$=require('my/express/1.0.0/my.express-1.0.0-model').$CCMM$;return _CTM$;}
ex$.$CCMM$=$CCMM$;
var m$1=require('ceylon/language/1.2.3/ceylon.language-1.2.3');
m$1.$addmod$(m$1,'ceylon.language/1.2.3');
m$1.$addmod$(ex$,'my.express/1.0.0');
ex$.$mod$ans$=function(){return[m$1.$_native(m$1.$arr$sa$(["js"],{t:m$1.$_String}))];};
function ExpressApp(expressApp$){
expressApp$.use$defs$handler=function($7,$8){return null;};
}
ExpressApp.dynmem$=['use','listen'];ExpressApp.$crtmm$=function(){return{mod:$CCMM$,pa:1,d:['my.express','ExpressApp']};};
ex$.ExpressApp=ExpressApp;
function $init$ExpressApp(){
if(ExpressApp.$$===undefined){
m$1.initTypeProtoI(ExpressApp,'my.express::ExpressApp');
(function(expressApp$){
expressApp$.use={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'path',mt:'prm',$t:{t:m$1.Anything}},{nm:'handler',mt:'prm',def:1,$t:{t:m$1.Anything}}],$cont:ExpressApp,pa:5,d:['my.express','ExpressApp','$m','use']};}};expressApp$.listen={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'port',mt:'prm',$t:{t:m$1.Integer}},{nm:'host',mt:'prm',$t:{t:m$1.$_String}},{nm:'handler',mt:'prm',$t:{t:m$1.Callable,a:{Return$Callable:{t:m$1.Anything},Arguments$Callable:{t:m$1.Empty}}}}],$cont:ExpressApp,pa:5,d:['my.express','ExpressApp','$m','listen']};}};
})(ExpressApp.$$.prototype);
}
return ExpressApp;
}
ex$.$init$ExpressApp=$init$ExpressApp;
$init$ExpressApp();
function MbaasExpress(mbaasExpress$){
}
MbaasExpress.dynmem$=['sys','mbaas','fhmiddleware','errorHandler'];MbaasExpress.$crtmm$=function(){return{mod:$CCMM$,pa:1,d:['my.express','MbaasExpress']};};
ex$.MbaasExpress=MbaasExpress;
function $init$MbaasExpress(){
if(MbaasExpress.$$===undefined){
m$1.initTypeProtoI(MbaasExpress,'my.express::MbaasExpress');
(function(mbaasExpress$){
mbaasExpress$.sys={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'endpoints',mt:'prm',$t:{t:m$1.Anything}}],$cont:MbaasExpress,pa:5,d:['my.express','MbaasExpress','$m','sys']};}};mbaasExpress$.$prop$getMbaas={$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},$cont:MbaasExpress,pa:5,d:['my.express','MbaasExpress','$at','mbaas']};}};
mbaasExpress$.fhmiddleware={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[],$cont:MbaasExpress,pa:5,d:['my.express','MbaasExpress','$m','fhmiddleware']};}};mbaasExpress$.errorHandler={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[],$cont:MbaasExpress,pa:5,d:['my.express','MbaasExpress','$m','errorHandler']};}};
})(MbaasExpress.$$.prototype);
}
return MbaasExpress;
}
ex$.$init$MbaasExpress=$init$MbaasExpress;
$init$MbaasExpress();
function Router(router$){
}
Router.dynmem$=['use','get','post'];Router.$crtmm$=function(){return{mod:$CCMM$,pa:1,d:['my.express','Router']};};
ex$.Router=Router;
function $init$Router(){
if(Router.$$===undefined){
m$1.initTypeProtoI(Router,'my.express::Router');
(function(router$){
router$.use={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'x',mt:'prm',$t:{t:m$1.Anything}}],$cont:Router,pa:5,d:['my.express','Router','$m','use']};}};router$.get={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'path',mt:'prm',$t:{t:m$1.$_String}},{nm:'f',mt:'prm',$t:{t:m$1.Callable,a:{Return$Callable:{t:m$1.Anything},Arguments$Callable:{t:'T',l:[{t:$init$Request()},{t:$init$Response()}]}}}}],$cont:Router,pa:5,an:function(){return[m$1.doc$($CCMM$,'my.express:Router:$m:get')];},d:['my.express','Router','$m','get']};}};router$.post={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'path',mt:'prm',$t:{t:m$1.$_String}},{nm:'f',mt:'prm',$t:{t:m$1.Callable,a:{Return$Callable:{t:m$1.Anything},Arguments$Callable:{t:'T',l:[{t:$init$Request()},{t:$init$Response()}]}}}}],$cont:Router,pa:5,d:['my.express','Router','$m','post']};}};
})(Router.$$.prototype);
}
return Router;
}
ex$.$init$Router=$init$Router;
$init$Router();
function Request(request$){
}
Request.dynmem$=['query','body'];Request.$crtmm$=function(){return{mod:$CCMM$,pa:1,d:['my.express','Request']};};
ex$.Request=Request;
function $init$Request(){
if(Request.$$===undefined){
m$1.initTypeProtoI(Request,'my.express::Request');
(function(request$){
})(Request.$$.prototype);
}
return Request;
}
ex$.$init$Request=$init$Request;
$init$Request();
function Response(response$){
}
Response.dynmem$=['json'];Response.$crtmm$=function(){return{mod:$CCMM$,pa:1,d:['my.express','Response']};};
ex$.Response=Response;
function $init$Response(){
if(Response.$$===undefined){
m$1.initTypeProtoI(Response,'my.express::Response');
(function(response$){
response$.json={$fml:1,$crtmm$:function(){return{mod:$CCMM$,$t:{t:m$1.Anything},ps:[{nm:'obj',mt:'prm',$t:{t:m$1.Anything}}],$cont:Response,pa:5,d:['my.express','Response','$m','json']};}};
})(Response.$$.prototype);
}
return Response;
}
ex$.$init$Response=$init$Response;
$init$Response();
ex$.$pkg$ans$my$express=function(){return[m$1.shared()];};
});
}(typeof define==='function' && define.amd ? define : function (factory) {
if (typeof exports!=='undefined') { factory(require, exports, module);
} else { throw 'no module loader'; }
}));
